apollo-router/src/protocols/sse/client.rs

```diff
@@ -0,0 +1,640 @@
+#![allow(dead_code)]
+#![allow(unreachable_pub)]
+
+use std::boxed;
+use std::error::Error as StdError;
+use std::fmt::Debug;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::fmt::{self};
+use std::future::Future;
+use std::io::ErrorKind;
+use std::pin::Pin;
+use std::str::FromStr;
+use std::task::Context;
+use std::task::Poll;
+use std::time::Duration;
+use std::time::Instant;
+
+use futures::ready;
+use futures::Stream;
+use hyper::body::HttpBody;
+use hyper::client::connect::Connect;
+use hyper::client::connect::Connection;
+pub use hyper::client::HttpConnector;
+use hyper::client::ResponseFuture;
+use hyper::header::HeaderMap;
+use hyper::header::HeaderName;
+use hyper::header::HeaderValue;
+use hyper::service::Service;
+use hyper::Body;
+use hyper::Request;
+use hyper::StatusCode;
+use hyper::Uri;
+use hyper_rustls::HttpsConnector as RustlsConnector;
+use hyper_timeout::TimeoutConnector;
+use pin_project_lite::pin_project;
+use tokio::io::AsyncRead;
+use tokio::io::AsyncWrite;
+use tokio::time::Sleep;
+
+use super::config::ReconnectOptions;
+use super::error::Error;
+use super::error::Result;
+use super::event_parser::EventParser;
+use super::event_parser::Sse;
+use super::retry::BackoffRetry;
+use super::retry::RetryStrategy;
+
+pub type HttpsConnector = RustlsConnector<HttpConnector>;
+
+type BoxError = Box<dyn std::error::Error + Send + Sync>;
+
+/// Represents a [`Pin`]'d [`Send`] + [`Sync`] stream, returned by [`Client`]'s stream method.
+pub type BoxStream<T> = Pin<boxed::Box<dyn Stream<Item = T> + Send + Sync>>;
+
+/// Client is the Server-Sent-Events interface.
+/// This trait is sealed and cannot be implemented for types outside this crate.
+pub(crate) trait Client: Send + Sync + private::Sealed {
+    fn stream(&self) -> BoxStream<Result<Sse>>;
+}
+
+/*
+ * TODO remove debug output
+ * TODO specify list of stati to not retry (e.g. 204)
+ */
+
+/// Maximum amount of redirects that the client will follow before
+/// giving up, if not overridden via [ClientBuilder::redirect_limit].
+pub const DEFAULT_REDIRECT_LIMIT: u32 = 16;
```

Comment:
```suggestion
pub(crate) const DEFAULT_REDIRECT_LIMIT: u32 = 16;
```

does it even need to be exported ?

Reply:
Yup, it's not exported anymore